src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -111,6 +113,34 @@ public LibraryServices(ProtoCore.Core libraryManagementCore, IPathManager pathMa
             PopulateOperators();
             PopulatePreloadLibraries();
             LibraryLoadFailed += new EventHandler<LibraryLoadFailedEventArgs>(LibraryLoadFailureHandler);
+
+            AppDomain.CurrentDomain.AssemblyResolve += ResolveAssembly;
+        }
+
+        private Assembly ResolveAssembly(object sender, ResolveEventArgs args)
+        {
+            var assemblyName = new AssemblyName(args.Name).Name + ".dll";
+            var assemblyPath = assemblyName;
+            try
+            {
+                if (pathManager.ResolveLibraryPath(ref assemblyPath))
+                {
+                    return Assembly.LoadFrom(assemblyPath);
+                }
+
+                foreach (var packagesDirectory in pathManager.PackagesDirectories)
```

Comment:
seems odd we need to do this - aren't the package directory /bin folders added to the assembly resolution paths?

on that note - why do we need a new assembly resolve handler?

Reply:
I don't think package directories are added to assembly resolution paths. Therefore the need to do this here.

Reply:
@aparajit-pratap what about right here:
https://github.com/DynamoDS/Dynamo/blob/f6f43e2a56655181bad6195a6bd0403a8e1b1a86/src/DynamoPackages/PackageLoader.cs#L222 ?

Reply:
So it looks like the existing assembly resolvers do not use these resolution paths.